### PR TITLE
fix(directory): fix un named directory

### LINKED ISSUE #775: bug(files): Folder will save as "un-named directory"
**OS:**
-Windows
-Linux
-MacOS

**Issue:**
-A user is able to save a folder with no name and it will appear with the name "un-named directory" 

**STR:**
-Go to Files / Create a folder
-Type some text into the input box 
-Backspace all your text
-Save folder

**Screenshot/recording:**

Screenshot 2023-05-02 at 9 22 26 PM

https://user-images.githubusercontent.com/93608357/235817570-36b82183-88cf-4d8f-a0be-0566154c833b.mov

### PR DESCRIPTION
**Pull request by lgmarchi**
### 1. Allow create new directory just with at least 1 char and less than 64 chars, added message validation
image

image

### 2. Added notification if user try a name that already exist when creating folder 

https://user-images.githubusercontent.com/63157656/235930440-64ba3564-15b6-457f-8675-f918fc5f63fb.mov

### 3. Get out create folder if user type Escape 

https://user-images.githubusercontent.com/63157656/235930579-db180d0e-f498-45ff-8aa4-d1cf5bedd53e.mov

### 4. Added notification if user try to rename a folder and the name already exist

https://user-images.githubusercontent.com/63157656/235930839-2c3a6e08-0c6b-47a5-9759-956100c61b6d.mov

### 5. Added notification if user try to rename a file and the name already exist

https://user-images.githubusercontent.com/63157656/235937264-eab53484-32a7-48c6-b2ec-3640dec91db8.mov

- 

### Which issue(s) this PR fixes 🔨

- Resolve #775

### CODE REVIEW

File: common/locales/en-US/main.ftl [hunk 1]
```diff
@@ -128,6 +128,7 @@ files = Files
     .doc-thumb = First Page Thumb
     .no-thumbnail-preview = No thumbnail file preview
     .file-already-opened = File already opened
+    .directory-alrady-with-name = There is already a directory with this name
```
Comment (stavares843) [inline common/locales/en-US/main.ftl]:
```suggestion
    .directory-already-with-name = There is already a directory with this name
```

File: ui/src/layouts/storage.rs [hunk 2]
```diff
@@ -651,6 +667,19 @@ pub fn FilesLayout(cx: Scope<Props>) -> Element {
                                     aria_label: dir.name(),
                                     with_rename: *is_renaming_map.read() == Some(key),
                                     onrename: move |(val, key_code)| {
+                                        if let Some(_) = directories_list.read().iter().find(|dir| dir.name() == val) {
+                                            state
+                                            .write()
+                                            .mutate(common::state::Action::AddToastNotification(
+                                                ToastNotification::init(
+                                                    "".into(),
+                                                    get_local_text("files.directory-alrady-with-name"),
```
Comment (stavares843) [inline ui/src/layouts/storage.rs]:
```suggestion
                                                    get_local_text("files.directory-already-with-name"),
```

**Comment by stavares843** [comment]
can we add singular/plural for 1 character/characters? 


image

**Comment by lgmarchi** [comment]
> can we add singular/plural for 1 character/characters?
> 
> image

Added!

**Comment by phillsatellite** [comment]
Hi @lgmarchi something very minor from my end, I noticed if you type some chars you can click out of naming the folder but if you have nothing typed and try to exit by clicking out it will keep you in naming folder


https://user-images.githubusercontent.com/93608357/235954180-b921d7fb-ea50-413b-a844-bc025083cee0.mov

File: ui/src/layouts/storage.rs [hunk 3]
```diff
@@ -761,9 +787,23 @@ pub fn FilesLayout(cx: Scope<Props>) -> Element {
                                                 }
                                             },
                                             onrename: move |(val, key_code)| {
+                                                let new_name: String = val;
+                                                if let Some(_) = files_list.read().iter().find(|file| file.name() == new_name) {
```
Comment (dariusc93) [inline ui/src/layouts/storage.rs]:
```suggestion
                                                if files_list.read().iter().any(|file| file.name() == new_name) {
```

File: ui/src/layouts/storage.rs [hunk 4]
```diff
@@ -651,6 +664,19 @@ pub fn FilesLayout(cx: Scope<Props>) -> Element {
                                     aria_label: dir.name(),
                                     with_rename: *is_renaming_map.read() == Some(key),
                                     onrename: move |(val, key_code)| {
+                                        if let Some(_) = directories_list.read().iter().find(|dir| dir.name() == val) {
```
Comment (dariusc93) [inline ui/src/layouts/storage.rs]:
```suggestion
                                        if directories_list.read().iter().any(|dir| dir.name() == val) {
```

File: ui/src/layouts/storage.rs [hunk 5]
```diff
@@ -597,6 +597,19 @@ pub fn FilesLayout(cx: Scope<Props>) -> Element {
                                 with_rename: true,
                                 onrename: |(val, key_code)| {
                                     let new_name: String = val;
+                                    if let Some(_) = directories_list.read().iter().find(|dir| dir.name() == new_name) {
```
Comment (dariusc93) [inline ui/src/layouts/storage.rs]:
```suggestion
                                    if directories_list.read().iter().any(|dir| dir.name() == new_name) {
```

**Review by dariusc93 — APPROVED** [review]
Left some suggestions but LGTM

**Comment by lgmarchi** [comment]
> Hi @lgmarchi something very minor from my end, I noticed if you type some chars you can click out of naming the folder but if you have nothing typed and try to exit by clicking out it will keep you in naming folder
> 
>  Screen_Recording_2023-05-03_at_10.47.25_AM.mov

Fixed my friend!

**Comment by phillsatellite** [comment]
One last very minor thing that only happens on Linux is when you exit naming the folder the screen will jump down a little 

https://user-images.githubusercontent.com/93608357/236014989-1bb59a3c-8d52-40e5-be43-786cb8c3b3f5.mov

**Comment by lgmarchi** [comment]
> One last very minor thing that only happens on Linux is when you exit naming the folder the screen will jump down a little
> 
>  Screen.Recording.2023-05-03.at.2.50.19.PM.mov

Not happen on Dev?

**Comment by phillsatellite** [comment]
> > One last very minor thing that only happens on Linux is when you exit naming the folder the screen will jump down a little
> > Screen.Recording.2023-05-03.at.2.50.19.PM.mov
> 
> Not happen on Dev?

Replied to Lucas on a DM but wanted to clarify on PR that it does not happen on Dev

**Comment by dariusc93** [comment]
> > > One last very minor thing that only happens on Linux is when you exit naming the folder the screen will jump down a little
> > > Screen.Recording.2023-05-03.at.2.50.19.PM.mov
> > 
> > 
> > Not happen on Dev?
> 
> Replied to Lucas on a DM but wanted to clarify on PR that it does not happen on Dev

Something like this could be made into its own separate issue unless Lucas wish to resolve it here.

**Comment by phillsatellite** [comment]
> > > > One last very minor thing that only happens on Linux is when you exit naming the folder the screen will jump down a little
> > > > Screen.Recording.2023-05-03.at.2.50.19.PM.mov
> > > 
> > > 
> > > Not happen on Dev?
> > 
> > 
> > Replied to Lucas on a DM but wanted to clarify on PR that it does not happen on Dev
> 
> Something like this could be made into its own separate issue unless Lucas wish to resolve it here.

I talked to Sara about that and we thought it would be ideal to not introduce any new bugs into the dev branch and avoid a new ticket in backlog but if it'll make it easier to fix it in another ticket then I can write it up real quick @lgmarchi @dariusc93

**Comment by lgmarchi** [comment]
> > > > One last very minor thing that only happens on Linux is when you exit naming the folder the screen will jump down a little
> > > > Screen.Recording.2023-05-03.at.2.50.19.PM.mov
> > > 
> > > 
> > > Not happen on Dev?
> > 
> > 
> > Replied to Lucas on a DM but wanted to clarify on PR that it does not happen on Dev
> 
> Something like this could be made into its own separate issue unless Lucas wish to resolve it here.

I would prefer in other issue as well. I tested it on Dev today and see if new folder or try to rename file happens on Dev too. 


https://user-images.githubusercontent.com/63157656/236254817-4668e899-7b87-4b11-bc22-2e7548337736.mov

**Review by phillsatellite — APPROVED** [review]

**Comment by dariusc93** [comment]
> > > > > One last very minor thing that only happens on Linux is when you exit naming the folder the screen will jump down a little
> > > > > Screen.Recording.2023-05-03.at.2.50.19.PM.mov
> > > > 
> > > > 
> > > > Not happen on Dev?
> > > 
> > > 
> > > Replied to Lucas on a DM but wanted to clarify on PR that it does not happen on Dev
> > 
> > 
> > Something like this could be made into its own separate issue unless Lucas wish to resolve it here.
> 
> I talked to Sara about that and we thought it would be ideal to not introduce any new bugs into the dev branch and avoid a new ticket in backlog but if it'll make it easier to fix it in another ticket then I can write it up real quick @lgmarchi @dariusc93

While I do agree about not introducing bugs, the issue you mention doesnt really impact functionality or the UI to a degree that requires it to be fixed here.

**Pull request merged**